sdk/tables/Azure.Data.Tables/src/QueryFilter.cs

```diff
@@ -0,0 +1,83 @@
+    /// by automatically replacing, quoting, and escaping interpolated parameters.
+    /// For more information, see <see href="https://docs.microsoft.com/rest/api/storageservices/querying-tables-and-entities#constructing-filter-strings">Constructing Filter Strings</see>.
+    /// </summary>
+    public static class QueryFilter
+    {
+        /// <summary>
+        /// Create an OData filter expression from an interpolated string.  The interpolated values will be quoted and escaped as necessary.
+        /// </summary>
+        /// <param name="filter">An interpolated filter string.</param>
+        /// <returns>A valid OData filter expression.</returns>
+        public static string Create(FormattableString filter)
+        {
+            if (filter == null) { return null; }
+
+            string[] args = new string[filter.ArgumentCount];
+            for (int i = 0; i < filter.ArgumentCount; i++)
+            {
+                args[i] = filter.GetArgument(i) switch
+                {
+                    // Null
+                    null => "null",
+
+                    // Boolean
+                    bool x => x.ToString(CultureInfo.InvariantCulture).ToLowerInvariant(),
+
+                    // Numeric
+                    sbyte x => x.ToString(CultureInfo.InvariantCulture),
+                    byte x => x.ToString(CultureInfo.InvariantCulture),
+                    short x => x.ToString(CultureInfo.InvariantCulture),
+                    ushort x => x.ToString(CultureInfo.InvariantCulture),
+                    int x => x.ToString(CultureInfo.InvariantCulture),
+                    uint x => x.ToString(CultureInfo.InvariantCulture),
+                    long x => x.ToString(CultureInfo.InvariantCulture),
+                    ulong x => x.ToString(CultureInfo.InvariantCulture),
+                    decimal x => x.ToString(CultureInfo.InvariantCulture),
+                    float x => x.ToString(CultureInfo.InvariantCulture),
+                    double x => x.ToString(CultureInfo.InvariantCulture),
+
+                    // Guid
+                    Guid x => $"{XmlConstants.LiteralPrefixGuid}'{x.ToString()}'",
+
+                    // binary
+                    byte[] x => $"X'{string.Join(string.Empty, x.Select(b => b.ToString("D2", CultureInfo.InvariantCulture)))}'",
+                    BinaryData x => $"X'{string.Join(string.Empty, x.ToArray().Select(b => b.ToString("D2", CultureInfo.InvariantCulture)))}'",
+
+                    // Dates as 8601 with a time zone
+                    DateTimeOffset x => $"{XmlConstants.LiteralPrefixDateTime}'{XmlConvert.ToString(x.UtcDateTime, XmlDateTimeSerializationMode.RoundtripKind)}'",
+                    DateTime x => $"{XmlConstants.LiteralPrefixDateTime}'{XmlConvert.ToString(x.ToUniversalTime(), XmlDateTimeSerializationMode.RoundtripKind)}'",
+
+                    // Text
+                    string x => $"'{x}'",
```

Comment:
Don't we need to do some escaping?

Reply:
The single quotes are the escaping. Do you mean url encode?

Reply:
What if the string contains single quotes?

Reply:
I don't think the goal is to protect against malformed queries, it's to eliminate the need to understand the escaping.

Reply:
>it's to eliminate the need to understand the escaping

Agreed. But you don't :) If my query is  `value = {s}` and s == `Pavel's` now I'm back to needing to understand the escaping.

Reply:
You are correct - I missed that case - will fix.